pkg/gadgets/traceloop/gadget.go

```diff
@@ -126,7 +126,7 @@ func (t *Trace) Start(trace *gadgetv1alpha1.Trace) {
 		# Indeed, traceloop creates a lof of tracers which each has maps and events:
 		# https://github.com/kinvolk/traceloop/blob/6f4efc6fca46d92c75f4ec4e6c6e1d829bdeaddf/bpf/straceback-guess-bpf.h#L27-L28
 		# So, this can generate a lof ot opened files.
-		ulimit -n unlimited
+		ulimit -n hard
```

Comment:
What is the current hard limit on the system where you have a problem?

I wonder if both the hard and soft limit should be changed instead:

```suggestion
		ulimit -Hn unlimited
		ulimit -Sn unlimited
```

Reply:
The current hard limit is 524288.
I cannot use unlimited:

```bash
root [ /home/renaud/inspektor-gadget/integration ]# kubectl exec -ti -n gadget gadget-5mk6h -- bash
root@cbl-mariner:/# ulimit -Sn
1024
root@cbl-mariner:/# ulimit -Hn
524288
root@cbl-mariner:/# ulimit -n
1024
root@cbl-mariner:/# ulimit -Sn unlimited
bash: ulimit: open files: cannot modify limit: Invalid argument
root@cbl-mariner:/# ulimit -Hn unlimited
bash: ulimit: open files: cannot modify limit: Operation not permitted
```

Hence, the use of `hard` (which is big enough to satisfy our needs).

Reply:
The default value of the hard limit can be configured differently on different systems & different container runtimes. Here they mention a default hard limit of 2560:
https://github.com/containers/common/blob/8eeaf9f5c09a212d14891cb8a0f2a3889d05ae4d/pkg/config/containers.conf#L79

In Podman 1.9, the hard limit used to be 1024 before being changed to 524288: https://github.com/containers/podman/issues/6093#issuecomment-624254340

So I am worried it might not be enough. But I don't have better suggestions for now. I suggest  to merge this patch and revisit if it causes problems.

Reply:
Clearly, it does not seem to have a silver bullet.
If we want to increase this limit, we can run the container as `privileged`, but we worked on this to restrain its privileges as much as possible so I do not want to choose this solution.

Anyway, I am not really worried about it: one day we will rework `traceloop` so this solution is only temporary.